File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        
        
        vector<int>ans;
        
        int i=0;
        int j=0;
        int size = nums.size();
        while(i<size and j<size){
            
            while(i<size and nums[i]<0) i++;
            
            while(j<size and nums[j]>0) j++;
            
            ans.push_back(nums[i]);
            ans.push_back(nums[j]);
            
            i++;
            j++;
        }
        
        return ans;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        
        p =0
        n =0
        li =[]
        size = len(nums)
        while p< size and n<size:
            
            while nums[p] <0: p+=1
            
            while nums[n]>0: n+=1
                
            li.append(nums[p])
            li.append(nums[n])
            p+=1
            n+=1
        return li
        